src/Models/Price.php

```diff
@@ -39,16 +53,70 @@ protected function casts(): array
     {
         return [
             'price' => Money::class,
+            'gross' => Money::class,
+            'net' => Money::class,
+            'discount_flat' => Money::class,
+            'discount_percentage' => Percentage::class,
+            'is_inherited' => 'boolean',
         ];
     }
 
+    public function basePrice(): Attribute
+    {
+        return Attribute::get(
+            fn () => $this->basePrice
+        );
+    }
+
     public function isNet(): Attribute
     {
         return Attribute::get(
             fn () => $this->priceList?->is_net
         );
     }
 
+    public function gross(): Attribute
+    {
+        return Attribute::get(
+            fn () => $this->getGross(data_get($this->product, 'vatRate.rate_percentage') ?: 0)
+        );
+    }
+
+    public function net(): Attribute
+    {
+        return Attribute::get(
+            fn () => $this->getNet(data_get($this->product, 'vatRate.rate_percentage') ?: 0)
+        );
+    }
+
+    public function isInherited(): Attribute
+    {
+        return Attribute::get(
+            fn () => $this->isInherited
+        );
+    }
+
+    public function discountPercentage(): Attribute
+    {
+        return Attribute::get(
+            fn () => $this->discountPercentage
+        );
+    }
+
+    public function discountFlat(): Attribute
+    {
+        return Attribute::get(
+            fn () => $this->discountFlat
+        );
+    }
+
+    public function appliedDiscounts(): Attribute
+    {
+        return Attribute::get(
+            fn () => collect($this->appliedDiscounts)->toArray()
```

Comment:
what does collect()->toArray() do? I suspect nothing.
Also appliedDiscounts is neither a property nor a relation